src/people.rs

```diff
@@ -789,6 +861,61 @@ pub trait ContextPeopleExt {
     /// `context.query_people(((Age, 30), (Gender, Female)))`.
     fn query_people<T: Query>(&self, q: T) -> Vec<PersonId>;
     fn match_person<T: Query>(&self, person_id: PersonId, q: T) -> bool;
+    fn get_counts<T: Tabulator, F>(&self, tabulator: &T, print_fn: F)
+    where
+        F: Fn(&Context, &[String], usize);
+}
+
+fn process_indices(
+    context: &Context,
+    remaining_indices: &[&Index],
+    accumulated_values: &mut [IndexValue],
+    display_values: &mut Vec<String>,
+    current_matches: &mut HashSet<PersonId>,
+    intersect: bool,
+    print_fn: &dyn Fn(&Context, &[String], usize),
+) {
+    if remaining_indices.is_empty() {
+        print_fn(context, display_values, current_matches.len());
+        return;
+    }
+
+    if let Some((next_index, rest_indices)) = remaining_indices.split_first() {
```

Comment:
`(next_index, rest_indices)) = remaining_indices.split_first().unwrap()`. This can't fail, so why have a conditional